models/articleid.model.js

```diff
@@ -0,0 +1,21 @@
+const db = require("../db/connection");
+
+const fetchArticleById = (article_id) => {
+  return db
+
+    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    
+    .then(({ rows }) => {
+        
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      }
+     if(article_id==="string"){
+        return Promise.reject({ status: 400, msg: "Bad Request" });
+     }
```

Comment:
The logic here (lines 13 to 15) is not needed as if article_id is wrong data type PSQL will send an error anyway, therefore you do not need to do it.